presto-rcfile/src/test/java/com/facebook/presto/rcfile/RcFileTester.java

```diff
@@ -568,6 +626,265 @@ private static RcFileReader createRcFileReader(TempFile tempFile, Type type, RcF
+                type.writeLong(blockBuilder, Float.floatToIntBits((Float) value));
+            }
+            else if (DOUBLE.equals(type)) {
+                type.writeDouble(blockBuilder, ((Number) value).doubleValue());
+            }
+            else if (VARCHAR.equals(type)) {
+                type.writeSlice(blockBuilder, Slices.utf8Slice((String) value));
+            }
+            else if (VARBINARY.equals(type)) {
+                type.writeSlice(blockBuilder, Slices.wrappedBuffer(((SqlVarbinary) value).getBytes()));
+            }
+            else if (DATE.equals(type)) {
+                long days = ((SqlDate) value).getDays();
+                type.writeLong(blockBuilder, days);
+            }
+            else if (TIMESTAMP.equals(type)) {
+                long millis = ((SqlTimestamp) value).getMillisUtc();
+                type.writeLong(blockBuilder, millis);
+            }
+            else {
+                String baseType = type.getTypeSignature().getBase();
+                if (ARRAY.equals(baseType)) {
+                    List<?> array = (List<?>) value;
+                    Type elementType = type.getTypeParameters().get(0);
+                    BlockBuilder arrayBlockBuilder = blockBuilder.beginBlockEntry();
+                    for (Object elementValue : array) {
+                        writeValue(elementType, arrayBlockBuilder, elementValue);
+                    }
+                    blockBuilder.closeEntry();
+                }
+                else if (MAP.equals(baseType)) {
+                    Map<?, ?> map = (Map<?, ?>) value;
+                    Type keyType = type.getTypeParameters().get(0);
+                    Type valueType = type.getTypeParameters().get(1);
+                    BlockBuilder mapBlockBuilder = blockBuilder.beginBlockEntry();
+                    for (Entry<?, ?> entry : map.entrySet()) {
+                        writeValue(keyType, mapBlockBuilder, entry.getKey());
+                        writeValue(valueType, mapBlockBuilder, entry.getValue());
+                    }
+                    blockBuilder.closeEntry();
+                }
+                else if (ROW.equals(baseType)) {
+                    List<?> array = (List<?>) value;
+                    List<Type> fieldTypes = type.getTypeParameters();
+                    BlockBuilder rowBlockBuilder = blockBuilder.beginBlockEntry();
+                    for (int fieldId = 0; fieldId < fieldTypes.size(); fieldId++) {
+                        Type fieldType = fieldTypes.get(fieldId);
+                        writeValue(fieldType, rowBlockBuilder, array.get(fieldId));
+                    }
+                    blockBuilder.closeEntry();
+                }
+                else {
+                    throw new IllegalArgumentException("Unsupported type " + type);
+                }
+            }
+        }
+    }
+
+    private static <K extends LongWritable, V extends BytesRefArrayWritable> void assertFileContentsOld(
```

Comment:
I skimmed through  this method and its callees. Let me know if you think this method needs careful review.